src/main/java/cinema/dao/Dao.java

```diff
@@ -0,0 +1,17 @@
+package cinema.dao;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface Dao<T> {
+
+    List<T> findAll();
+
+    Optional<T> findById(Long id);
+
+    void delete(Long id);
+
+    void update(T entity);
+
+    void save(T entity);
```

Comment:
may be better get T instead void

Reply:
fixed

Reply:
T update(T entity);

    T save(T entity);